pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -151,7 +152,9 @@ func (d *DependenciesDistributor) OnUpdate(oldObj, newObj interface{}) {
 		klog.V(4).Infof("Ignore update event of object (%s, kind=%s, %s) as specification no change", unstructuredOldObj.GetAPIVersion(), unstructuredOldObj.GetKind(), names.NamespacedKey(unstructuredOldObj.GetNamespace(), unstructuredOldObj.GetName()))
 		return
 	}
-	d.OnAdd(oldObj)
+	if !equality.Semantic.DeepEqual(unstructuredOldObj.GetLabels(), unstructuredNewObj.GetLabels()) {
```

Comment:
Can `DeepEqual` be used to determine whether two maps are the same? If so, can you help add the corresponding ut?

Reply:
Add  test case in  Test_OnUpdate

Reply:
What if there are more than two elements in the map?

Reply:
> What if there are more than two elements in the map?

add new test case